sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/PythonCoverageSensor.java

```diff
@@ -59,99 +59,68 @@ public void execute(SensorContext context, Map<InputFile, Set<Integer>> linesOfC
     String baseDir = context.fileSystem().baseDir().getPath();
     Settings settings = context.settings();
 
-    LOG.info("Python unit test coverage");
-    List<File> reports = getReports(settings, baseDir, REPORT_PATH_KEY, DEFAULT_REPORT_PATH);
-    Map<InputFile, NewCoverage> coverageMeasures = parseReports(reports, context);
-    HashSet<InputFile> filesCoveredByUT = new HashSet<>();
-    saveMeasures(coverageMeasures, filesCoveredByUT, CoverageType.UNIT);
-
-    LOG.info("Python integration test coverage");
-    List<File> itReports = getReports(settings, baseDir, IT_REPORT_PATH_KEY, IT_DEFAULT_REPORT_PATH);
-    Map<InputFile, NewCoverage> itCoverageMeasures = parseReports(itReports, context);
-    HashSet<InputFile> filesCoveredByIT = new HashSet<>();
-    saveMeasures(itCoverageMeasures, filesCoveredByIT, CoverageType.IT);
-
-    LOG.info("Python overall test coverage");
-    List<File> overallReports = getReports(settings, baseDir, OVERALL_REPORT_PATH_KEY, OVERALL_DEFAULT_REPORT_PATH);
-    Map<InputFile, NewCoverage> overallCoverageMeasures = parseReports(overallReports, context);
-    HashSet<InputFile> filesCoveredOverall = new HashSet<>();
-    saveMeasures(overallCoverageMeasures, filesCoveredOverall, CoverageType.OVERALL);
-
-    if (settings.getBoolean(FORCE_ZERO_COVERAGE_KEY)) {
+    HashSet<InputFile> filesCovered = new HashSet<>();
+    List<File> reports = new ArrayList<>();
+    reports.addAll(getReports(settings, baseDir, REPORT_PATH_KEY, DEFAULT_REPORT_PATH));
+    reports.addAll(getReports(settings, baseDir, IT_REPORT_PATH_KEY, DEFAULT_REPORT_PATH));
+    reports.addAll(getReports(settings, baseDir, OVERALL_REPORT_PATH_KEY, DEFAULT_REPORT_PATH));
+    if (!reports.isEmpty()) {
+      LOG.info("Python test coverage");
+      for (File report : reports) {
+        Map<InputFile, NewCoverage> coverageMeasures = parseReport(report, context);
+        saveMeasures(coverageMeasures, filesCovered);
+      }
+    }
+    if (settings.getBoolean(FORCE_ZERO_COVERAGE_KEY) || (!settings.hasKey(FORCE_ZERO_COVERAGE_KEY) && !reports.isEmpty())) {
```

Comment:
Why do we add this new condition?
```(!settings.hasKey(FORCE_ZERO_COVERAGE_KEY) && !reports.isEmpty())```

Reply:
Removed from here and https://jira.sonarsource.com/browse/SONARPY-246